server/routes/referrals/completionDeadlinePresenter.ts

```diff
@@ -45,20 +50,14 @@ export default class CompletionDeadlinePresenter {
       this.year = String(userInputData['completion-deadline-year'] || '')
     }
 
-    if (!errors) {
-      this.errorSummary = null
-      this.errorMessage = null
-      this.erroredFields = []
-    } else {
-      this.errorSummary = {
-        errors: [{ message: errors.message, linkedField: errors.firstErroredField }],
-      }
-      this.errorMessage = errors.message
-      this.erroredFields = errors.erroredFields
-    }
-  }
-}
+    this.errorSummary = ReferralDataPresenterUtils.errorSummary(error)
+    this.errorMessage =
+      ReferralDataPresenterUtils.errorMessage(error, 'completion-deadline-day') ??
+      ReferralDataPresenterUtils.errorMessage(error, 'completion-deadline-month') ??
+      ReferralDataPresenterUtils.errorMessage(error, 'completion-deadline-year')
 
-export interface CompletionDeadlineErrorSummaryPresenter {
-  errors: { message: string; linkedField: 'day' | 'month' | 'year' }[]
+    this.hasDayError = !!error?.errors.find(subError => subError.formFields.includes('completion-deadline-day'))
```

Comment:
Not a huge deal but would it be possible to wrap this in a nicely named function to make it quicker to parse and reduce repetition in the next three lines e.g. `hasError(error: FormValidationError | null, fieldName: string): boolean`

Reply:
That's a good shout - added it.